astropy/timeseries/downsample.py

```diff
@@ -13,13 +13,35 @@
 __all__ = ["aggregate_downsample"]
 
 
+def nanmean_reduceat(data, indices):
+    mask = np.isnan(data)
+
+    if mask.any():  # If there are NaNs
+        # Create a writeable copy and mask NaNs
+        data = data.copy()
+        data[mask] = 0
+        count_data = np.add.reduceat(~mask, indices)
+        count_data = count_data.astype(float)
```

Comment:
`float` generally maps to `np.float64` *except* for numpy 1.x on Windows (where it gives `float32`). It may be worth using the explicitly portable form to avoid the pain of debugging this to anyone who might encounter a portability issue later.

```suggestion
        count_data = count_data.astype("float64")
```